paperbackup-restore.sh

```diff
@@ -1,33 +1,38 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 # restore data backed up with paperbackup.py
 
 # give one file containing all qrcodes as parameter
 
-SCANNEDFILE=$1
+SCANNEDFILE="$1"
+
+if [ "$(uname)" = "Darwin" ]; then
+    PATH="/usr/local/bin:$PATH"
+    alias sed="gsed"
+fi
 
 if [ -z "$SCANNEDFILE" ]; then
     echo "give one file containing all qrcodes as parameter"
     exit 1
 fi
 
-if ! [ -f "$SCANNEDFILE" ]; then
-    echo "$SCANNEDFILE is not a file"
+if [ ! -f "$SCANNEDFILE" ]; then
+    echo "$SCANNEDFILE is not a file" > /dev/stderr
     exit 1
 fi
 
-if [ ! -x "/usr/bin/zbarimg" ]; then
-    echo "/usr/bin/zbarimg missing"
+which zbarimg > /dev/null 2>&1 || {
+    echo "zbarimg not found in PATH" > /dev/stderr
     exit 2
-fi
+}
```

Comment:
i this case i wouldn't bother autodetecting or even warning, you'll have a good error just by calling `zbarimg` without a path prefix lower down.